### PR TITLE
Fix mysql version and travis mysql build

### PR DESCRIPTION
**Pull request by pascalgrimaud**
I fix 2 issues here:

### ZonedDateTime problem with MySQL 5.7.14

In https://github.com/jhipster/generator-jhipster/issues/4009, @tsaqova mentionned

> here are my schema, i change mysql version to 5.7.13 as workaround because 5.7.14 can't have 2 timestamp column (from zonedatetime) in one table by default

Here the error log on Travis build:

```
liquibase.exception.DatabaseException: Invalid default value for 'zoned_date_time_required_field' [Failed SQL: CREATE TABLE travisMysql.field_test_entity (id BIGINT AUTO_INCREMENT NOT NULL, string_field VARCHAR(255) NULL, string_required_field VARCHAR(255) NOT NULL, string_minlength_field VARCHAR(255) NULL, string_maxlength_field VARCHAR(20) NULL, string_pattern_field VARCHAR(255) NULL, integer_field INT NULL, integer_required_field INT NOT NULL, integer_min_field INT NULL, integer_max_field INT NULL, long_field BIGINT NULL, long_required_field BIGINT NOT NULL, long_min_field BIGINT NULL, long_max_field BIGINT NULL, float_field FLOAT NULL, float_required_field FLOAT NOT NULL, float_min_field FLOAT NULL, float_max_field FLOAT NULL, double_required_field DOUBLE NOT NULL, double_min_field DOUBLE NULL, double_max_field DOUBLE NULL, big_decimal_required_field DECIMAL(10, 2) NOT NULL, big_decimal_min_field DECIMAL(10, 2) NULL, big_decimal_max_field DECIMAL(10, 2) NULL, local_date_field date NULL, local_date_required_field date NOT NULL, zoned_date_time_field timestamp NULL, zoned_date_time_required_field timestamp NOT NULL, boolean_field BIT NULL, boolean_required_field BIT NOT NULL, enum_field VARCHAR(255) NULL, enum_required_field VARCHAR(255) NOT NULL, byte_image_field LONGBLOB NULL, byte_image_field_content_type VARCHAR(255) NULL, byte_image_required_field LONGBLOB NOT NULL, byte_image_required_field_content_type VARCHAR(255) NOT NULL, byte_image_minbytes_field LONGBLOB NULL, byte_image_minbytes_field_content_type VARCHAR(255) NULL, byte_image_maxbytes_field LONGBLOB NULL, byte_image_maxbytes_field_content_type VARCHAR(255) NULL, byte_any_field LONGBLOB NULL, byte_any_field_content_type VARCHAR(255) NULL, byte_any_required_field LONGBLOB NOT NULL, byte_any_required_field_content_type VARCHAR(255) NOT NULL, byte_any_minbytes_field LONGBLOB NULL, byte_any_minbytes_field_content_type VARCHAR(255) NULL, byte_any_maxbytes_field LONGBLOB NULL, byte_any_maxbytes_field_content_type VARCHAR(255) NULL, byte_text_field LONGTEXT NULL, byte_text_required_field LONGTEXT NOT NULL, byte_text_minbytes_field LONGTEXT NULL, byte_text_maxbytes_field LONGTEXT NULL, CONSTRAINT PK_FIELD_TEST_ENTITY PRIMARY KEY (id))]
	at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:301)
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:55)
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:107)
	at liquibase.database.AbstractJdbcDatabase.execute(AbstractJdbcDatabase.java:1251)
	at liquibase.database.AbstractJdbcDatabase.executeStatements(AbstractJdbcDatabase.java:1234)
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:554)
	at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:51)
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:73)
	at liquibase.Liquibase.update(Liquibase.java:212)
	at liquibase.Liquibase.update(Liquibase.java:192)
```

I simply use MySQL 5.7.13 instead of 5.7.14.
When I have time, I'll try to investigate to find if there is a specific environment variable to put in docker-compose file for mysql or something else.

### ConstraintName for MySQL and Oracle

Here the error log on Travis build:

```
2016-08-25 07:52:34.383 ERROR 6726 --- [           main] liquibase                                : classpath:config/liquibase/master.xml: classpath:config/liquibase/changelog/20160208210534_added_entity_constraints_TestManyToMany.xml::20160208210534-2::jhipster: Change Set classpath:config/liquibase/changelog/20160208210534_added_entity_constraints_TestManyToMany.xml::20160208210534-2::jhipster failed.  Error: Identifier name 'test_many_to_many_test_custom_table_name_test_custom_table_names_id' is too long [Failed SQL: ALTER TABLE travisMysql.test_many_to_many_test_custom_table_name ADD CONSTRAINT test_many_to_many_test_custom_table_name_test_custom_table_names_id FOREIGN KEY (test_custom_table_names_id) REFERENCES travisMysql.test_custom_table_name_entity (id)]
liquibase.exception.DatabaseException: Identifier name 'test_many_to_many_test_custom_table_name_test_custom_table_names_id' is too long [Failed SQL: ALTER TABLE travisMysql.test_many_to_many_test_custom_table_name ADD CONSTRAINT test_many_to_many_test_custom_table_name_test_custom_table_names_id FOREIGN KEY (test_custom_table_names_id) REFERENCES travisMysql.test_custom_table_name_entity (id)]
	at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:301)
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:55)
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:107)
	at liquibase.database.AbstractJdbcDatabase.execute(AbstractJdbcDatabase.java:1251)
	at liquibase.database.AbstractJdbcDatabase.executeStatements(AbstractJdbcDatabase.java:1234)
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:554)
	at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:51)
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:73)
	at liquibase.Liquibase.update(Liquibase.java:212)
	at liquibase.Liquibase.update(Liquibase.java:192
```

It happens when on the code, `noSnakeCase` is true and `entityName + relationshipName + '_id'` is too long
Don't hesitate to change it if someone has a better idea for that.

### CODE REVIEW

File: generators/generator-base.js [hunk 1]
```diff
@@ -1196,7 +1196,15 @@ Generator.prototype.getConstraintName = function (entityName, relationshipName,
         var halfLimit = Math.floor(limit/2),
             entityTable = noSnakeCase ? entityName : this.getTableName(entityName.substring(0, halfLimit)),
             relationTable = noSnakeCase ? relationshipName: this.getTableName(relationshipName.substring(0, halfLimit - 1));
-        return `${entityTable}_${relationTable}_id`;
+
+        constraintName = `${entityTable}_${relationTable}_id`;
```
Comment (deepu105) [inline generators/generator-base.js]: the correct fix here should be changing `entityTable = noSnakeCase ? entityName : this.getTableName(entityName.substring(0, halfLimit))` to `entityTable = noSnakeCase ? entityName.substring(0, halfLimit) : this.getTableName(entityName.substring(0, halfLimit))` and changing `relationTable = noSnakeCase ? relationshipName: this.getTableName(relationshipName.substring(0, halfLimit - 1));` to `relationTable = noSnakeCase ? relationshipName.substring(0, halfLimit - 1) : this.getTableName(relationshipName.substring(0, halfLimit - 1));`

Comment (deepu105) [inline generators/generator-base.js]: can you remove this part from the PR? ill fix it in master

Comment (deepu105) [inline generators/generator-base.js]: its done https://github.com/jhipster/generator-jhipster/commit/9ff9e1cba9112e9242bc2cb65594aae34b08ccf3

**Comment by pascalgrimaud** [comment]
I can't do it until tonight. If you can do it, plz feel free. Simply change docker version

**Comment by deepu105** [comment]
Ok ill merge it manually

Thanks & regards,
Deepu

**Pull request merged**